convert2rhel/actions/__init__.py

```diff
@@ -206,25 +211,94 @@ def run(self):
+        :type id: str
+        :param level: The status_code of the result .
+        :type: level: str
+        :param message: The message to be set.
+        :type message: str | None
+        """
+        if level not in ("ERROR", "OVERRIDABLE", "SKIP", "SUCCESS"):
+            raise KeyError("The level of result must be FAILURE, OVERRIDABLE, SKIP, or SUCCESS.")
+
+        result = ActionResult(level, id, message)
+        self.result = result
+
+    def add_message(self, level, id=None, message=None):
         """
-        Helper method that sets the resulting values for status, error_id and message.
+        Helper method that adds the resulting values for level, id and message of a warning or info log message.
 
-        :param status: The status to be set.
-        :type: status: str
-        :param error_id: The error_id to identify the error.
-        :type error_id: str
+        :param id: The id to identify the message.
+        :type id: str
+        :param level: The level to be set.
+        :type: level: str
         :param message: The message to be set.
         :type message: str | None
         """
-        if status != _NO_USER_VALUE:
-            self.status = STATUS_CODE[status]
+        msg = ActionMessage(level, id, message)
+        self.messages.append(msg)
 
-        if error_id != _NO_USER_VALUE:
-            self.error_id = error_id
 
-        if message != _NO_USER_VALUE:
-            self.message = message
+class ActionMessageBase:
+    def __init__(self, level="SUCCESS", id=None, message=""):
+        self.id = id
+        self.level = STATUS_CODE[level]
+        self.message = message
+
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "level": self.level,
+            "message": self.message,
+        }
+
+
+class ActionMessage(ActionMessageBase):
+    def __init__(self, level, id=None, message=""):
+        if not (id and level and message):
+            raise InvalidMessage("Messages require id, level and message fields")
+
+        if not (STATUS_CODE["SUCCESS"] < STATUS_CODE[level] < STATUS_CODE["SKIP"]):
+            raise InvalidMessage("Invalid level '%s', set for a non-result message" % level)
+
+        super(ActionMessage, self).__init__(level, id, message)
+
+
+class ActionResult(ActionMessageBase):
+    def __init__(self, level="SUCCESS", id=None, message=""):
+
+        if STATUS_CODE[level] >= STATUS_CODE["SKIP"]:
+            if not id and not message:
+                raise InvalidMessage("Non-success results require an id and a message")
+
+            if not id:
+                raise InvalidMessage("Non-success results require an id")
+
+            if not message:
+                raise InvalidMessage("Non-success results require a message")
+
+        elif STATUS_CODE["SUCCESS"] < STATUS_CODE[level] < STATUS_CODE["SKIP"]:
+            raise InvalidMessage("The level for result must be SKIP or more fatal or SUCCESS.")
```

Comment:
The message here is kind of confusing... It is not clear if the level should be between the levels `SUCCESS` and `SKIP`, or , if it is only allowed `SUCCESS` or anything higher than `SKIP`. 

I would suggest doing the same as the previous class, both in the if statement as the message, to make it more explicit of the intentions here. 

If this will be changed, I would suggest creating a method in `ActionMessageBase` that does this validation, as it is something common for both classes that inherit it.

Reply:
This message is saying that the level cannot be anything between success and skip, in other words no WARNING or INFO messages (if we add that)

Reply:
So for example if we add a result with a warning level then this error will be raised to say the level needs to be either SKIP, more fatal than SKIP or SUCCESS

Reply:
I can add what level caused the error